File: leetcode2/second/0669. Trim a Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def trimBST(self, root, L, R):
        """
        :type root: TreeNode
        :type L: int
        :type R: int
        :rtype: TreeNode
        """
        if not root:
            return root
        if R<root.val:
            return self.trimBST(root.left,L,R)
        elif L>root.val:
            return self.trimBST(root.right,L,R)
        else:
            root.left=self.trimBST(root.left,L,root.val)
            root.right=self.trimBST(root.right,root.val,R)
            return root

File: archives/C#/0669. Trim a Binary Search Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode TrimBST(TreeNode root, int L, int R) {
        if(root==null){
            return root;
        }
        else if(root.val>R){
            return TrimBST(root.left,L,R);
        }
        else if(root.val<L){
             return TrimBST(root.right,L,R);
        }
        else{
            root.left=TrimBST(root.left,L,R);
            root.right=TrimBST(root.right,L,R);
            return root;
        }
    }
}
